src/lib/crypto/kimchi_bindings/js/native/native-overrides/oracles.js

```diff
@@ -0,0 +1,38 @@
+/* global plonk_wasm, tsRustConversionNative,*/
+
+
+// Provides: caml_pasta_fq_poseidon_block_cipher
+// Requires: plonk_wasm, tsRustConversionNative
+function caml_pasta_fq_poseidon_block_cipher(_fake_params, fq_vector) {
+
+    console.log("overriding the old wasm caml_pasta_fq_poseidon_block_cipher, now using the native one")
+
+  // 1. get permuted field vector from rust
+  var wasm_flat_vector = plonk_wasm.caml_pasta_fq_poseidon_block_cipher(
+    tsRustConversionNative.fq.vectorToRust(fq_vector)
```

Comment:
It is nice that you could make this `vectorToRust` work! Luckily I was working on the others inside the `conversionCore` (meaning `gate/point/polyCommTo/FromRust`), so we didn't step on each other's lines of work 👍🏻